docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+```
+git checkout master
+```
+
+Now we need to rebase the master branch so that it matches the upstream master branch.
+
+```
+git pull --rebase upstream master
+```
+
+Rebase is destructive and will rewrite your current master branch. This is fine because we always want the master branch to be identical to the upstream repository's master branch.
+
+By default, Git may reject the update because it will detect that the changes are due to a rebase, so it is necessary to `force` the update.
+
+```
+git push -f origin master
+```
+
+Next we need to check out the feature branch.
+
+```
+git checkout { feature branch }
+
+```
+
+In my case:
+
+```
+git checkout emonddr-doc-changes
+```
+
+
+ Now, rebase the local feature branch with the local copy of the master branch.
+
+```
+git rebase master
+```
+
+Next we need to push the local feature branch to the remote feature branch.
+
+```
+git push -f origin { feature branch }
+```
+
+In my case:
+
+```
+git push -f origin emonddr-doc-changes
+```
+
+Now we can see that the copy of the repository is no longer behind in several commits.
+
+The `master` branch the forked repository is `even` with original repository.
+
+![submit_pr_rebase_3.png](./imgs/submit_pr_rebase_3.png)
+
+
+The `feature` branch of the forked repository is 1 commit ahead of the `master` branch, and not `behind` on any number of commits.
+
+![submit_pr_rebase_2.png](./imgs/submit_pr_rebase_2.png)
+
+
+### 16. Squashing Commits
+
+Squashing commits is the process of compressing many commits into one, and is a great way to keep repositories concise and clean. It is also a necessary step before actually creating a pull request.
+
+In my case, I have 3 commits in my remote feature branch that I need to squash into one.
+
+![submit_pr_squash_commits_1.png](./imgs/submit_pr_squash_commits_1.png)
+
+Before squashing commits, ensure that you have performed the `rebase` steps described in the previous section.
+
+Ensure that you currently have your feature branch checked out.
+
+Type :
+
+```
+git status
+```
+
+![submit_pr_squash_commits_2.png](./imgs/submit_pr_squash_commits_2.png)
+
+I do have my feature branch checked out.
+
+Now let's rebase the feature branch off of the master branch in an interactive mode that allows us to `squash` the commits.
+```
+git rebase -i master
+```
+
+![submit_pr_squash_commits_3.png](./imgs/submit_pr_squash_commits_3.png)
+
+The `vim` editor opens and lists my 3 commits, and some commands that are available.
+
+Type `i` to place the editor into `INSERT` mode.
+
+![submit_pr_squash_commits_4.png](./imgs/submit_pr_squash_commits_4.png)
+
+I want to keep the commit message from the first commit, so I will leave the word `pick` in front of the first commit, and change the words from `pick` to `squash` for the second and third commits. (Use the `arrow` keys to move around, and the `delete` or `backspace` key for deleting characters)
+
+![submit_pr_squash_commits_5.png](./imgs/submit_pr_squash_commits_5.png)
```

Comment:
A nitpick for the commands: you can a single letter for short, like
```
p 88bc27c5 docs: some description
s 67ac02a7 docs: some other description
```

Reply:
hmm, for beginners/intermediates... I prefer to put full words. :)